src/dosext/mfs/mfs.c

```diff
@@ -2545,14 +2546,28 @@ static int GetRedirection(struct vm86_regs *state)
 	/* set the high bit of the return CL so that */
 	/* NetWare shell doesn't get confused */
 	returnCX = drives[dd].user_param | 0x80;
-	returnDX = drives[dd].options;
-
 	Debug0((dbg_fd, "GetRedirection CX=%04x\n", returnCX));
 
 	userStack = (u_short *) sda_user_stack(sda);
+
+        if (WORD(state->ecx) == REDIR_CLIENT_SIGNATURE) {
```

Comment:
Isn't this check redundant?
I think `DX,BP destroyed` in RBIL says that
it is. Also it seems the indentation is wrong?

Reply:
> Isn't this check redundant?

Yes, if you can believe that no client ever expects DX to be preserved.

> Also it seems the indentation is wrong?

Unfortunately the function is visually indented by 2 spaces, but there are many lines I didn't change where tabs are represent 8 spaces. So if whatever viewer you are using is not set to 8 space tabs it will look wrong. The real fix is to reindent it all, obviously.

Reply:
1 - DX is now used to return dosemu specific data regardless of what client is calling.

2 - I've now reindented the function using spaces in a prior commit, it should no look strange when looking at the changes.

3 - I expanded the CDROM unit width to three bits, but I think we need to add some checks in the lredir client as it doesn't seem to be validated.